### DISCUSSION TITLE
List of all `setting.git_savvy.XXX`

### DISCUSSION THREAD

**Discussion by igorolhovskiy** [timbrel/GitSavvy#1919] [Q&A]
Hello!

I have a bit strange request, but I want to some keybindings to behave differently based on a branch name. 
For this I can use "context", but not sure how to get current branch name from Sublime / GitSavvy.
Is there any way to get this info?

Cheers,
Ihor

**Comment by kaste** [comment]
What have you in mind?   Because it depends; maybe you need a short and perhaps trivial user plugin.

**Comment by igorolhovskiy** [comment]
Idea is the following:

Due to my workflow, I can test my code fully only if it's pushed to git. Due to this I need to do a commit / push a lot during the day. This I'm doing on my "personal" branch. 
After I'm happy with a result, I'm doing squash to "main", so it remains clean.

So, I made a trivial keybindings chain, mainly `git add -A -> git commit -m 'new try' -> git push` to save time. But I'd like to have a failsafe, that this keybinding will work only on my personal branch (which is holding my username on it), so it should work only if branch is matching regex.

Cheers,
Ihor

**Reply by kaste** [reply]
How does this "trivial" keybindings chain look like?  Do you trigger that from one of the views?  Or is it a global keybinding?  The idea of having a context key set to the current branch name is very interesting.  I haven't done that ... like ever.

**Reply by igorolhovskiy** [reply]
Sure, here is the chain

```
[
    {
        "keys": ["ctrl+f5", "ctrl+f6"],
        "command": "chain" ,
        "args": {
            "commands": [
                ["gs_custom", {"args": ["add", "-A"]}],
                ["gs_custom", {"args": ["commit", "-m", "[AUTO] Commit from Sublime Text"]}],
                ["gs_push"],
            ]
        }
    },
]
```
I've made it for now 2-keys sequence for have some "time to think", maybe need to change it to less "easy one".

**Reply by kaste** [reply]
This kind of chain. Okay.  Does the chain actually break when one of the commands fails? Or are these safe, cant-fail commands.  🤔 
I do like the context-key approach but don't have it at hand.  You could use this or something like this (aka: untested).

```
from GitSavvy.core.base_commands import GsWindowCommand
from GitSavvy.core import store


class add_n_push(GsWindowCommand):
    def run(self) -> None:
        state = store.current_state(self.repo_path)
        if head := state.get("head"):
            if head.branch != "<branch_name>":
                return

        self.add_all_files()
        self.git("commit", "-m", "[AUTO] Commit from Sublime Text")
        self.window.run_command("gs_push")
```

Safe the code in the user folder as a python file, e.g. `addnpush.py`.  And use `add_n_push` as the new command for the key binding.  

Can you actually write a context query key, t.i. iirc implement `on_query_context(...)`?

**Reply by igorolhovskiy** [reply]
Hm...

Thanks!
Will give a try with your code, maybe just will add some settings to control it, like `<branch_name>` will be taken from settings.

Cheers,
Ihor

**Reply by kaste** [reply]
Sure. 

Reading my code again I would probably recommend to just use raw `git` calls because these will not break when I refactor some stuff in GitSavvy.  E.g.

```
        self.git("add", "-A")
        self.git("commit", "-m", "[AUTO] Commit from Sublime Text")
        self.git("push")
```
